src/test/java/uk/gov/digital/ho/pttg/api/ArchiveResourceTest.java

```diff
@@ -77,10 +86,69 @@ public void archiveResult_logsSuccess() {
             String expectedLogMessage = "ArchiveResult request completed successfully";
             return loggingEvent.getFormattedMessage().equals(expectedLogMessage) &&
                     loggingEvent.getArgumentArray()[0].equals(new ObjectAppendingMarker("event_id", PTTG_AUDIT_ARCHIVE_RESULT_RESPONSE_SUCCESS)) &&
-                    ((ObjectAppendingMarker)loggingEvent.getArgumentArray()[1]).getFieldName().equals("request_duration_ms");
+                    ((ObjectAppendingMarker) loggingEvent.getArgumentArray()[1]).getFieldName().equals("request_duration_ms");
         }));
     }
 
+    @Test
+    public void getArchivedResults_givenFromDate_passedToService() {
+        LocalDate fromDate = LocalDate.of(2019, Month.JANUARY, 1);
+        archiveResource.getArchivedResults(fromDate, SOME_DATE);
+
+        verify(mockArchiveService).getArchivedResults(eq(fromDate), any(LocalDate.class));
+    }
+
+    @Test
+    public void getArchivedResults_givenToDate_passedToService() {
+        LocalDate toDate = LocalDate.of(2019, Month.JANUARY, 31);
+        archiveResource.getArchivedResults(SOME_DATE, toDate);
+
+        verify(mockArchiveService).getArchivedResults(any(LocalDate.class), eq(toDate));
+    }
+
+    @Test
+    public void getArchivedResults_resultsReturnedFromService_returnedByResource() {
+        List<ArchivedResult> archivedResults = singletonList(SOME_ARCHIVED_RESULT);
+        when(mockArchiveService.getArchivedResults(SOME_DATE, SOME_DATE))
+                .thenReturn(archivedResults);
+
+        assertThat(archiveResource.getArchivedResults(SOME_DATE, SOME_DATE))
+                .isEqualTo(archivedResults);
+    }
+
+    @Test
+    public void getArchivedResults_givenParameters_logRequest() {
+        LocalDate fromDate = LocalDate.of(2018, Month.DECEMBER, 1);
+        LocalDate toDate = LocalDate.of(2018, Month.DECEMBER, 31);
+
+        archiveResource.getArchivedResults(fromDate, toDate);
+
+        assertInfoLog("Request received for archived results between 2018-12-01 and 2018-12-31", PTTG_AUDIT_GET_ARCHIVED_RESULTS_REQUEST_RECEIVED);
+    }
+
+    @Test
+    public void getArchivedResults_givenResponse_logResponse() {
+        when(mockArchiveService.getArchivedResults(SOME_DATE, SOME_DATE))
+                .thenReturn(asList(
+                        SOME_ARCHIVED_RESULT,
+                        SOME_ARCHIVED_RESULT,
+                        SOME_ARCHIVED_RESULT
+                ));
+
+        archiveResource.getArchivedResults(SOME_DATE, SOME_DATE);
+
+        assertInfoLog("Returned response with 3 archived results", PTTG_AUDIT_GET_ARCHIVED_RESULTS_RESPONSE_SUCCESS);
+    }
+
+    public void assertInfoLog(String expectedMessage, LogEvent logEvent) {
```

Comment:
Can the "archiveResult_logs*" tests use this helper method?

Reply:
I've done archiveResult_logsRequestParameters but not archiveResult_logsSuccess as the latter checks for request_duration_ms in the argument array too.